### ISSUE TITLE
Using Multiple Formatters

### ISSUE THREAD

**Issue by adam-moss** [DavidAnson/markdownlint-cli2#84]
Hi

I'm just migrating across to `markdownlint-cli2` and am probably misunderstanding something.

If I define multiple formatters:

```yaml
outputFormatters:
    - - markdownlint-cli2-formatter-default
      - markdownlint-cli2-formatter-junit
```

I was expecting to get the output twice, once to the terminal and once to the `markdownlint-cli2-junit.xml` file.

However I only get the output once, either to the terminal or to the file depending on the order they're listed in the config.

Is this expected behaviour or am I missing something?

Thanks

**Comment by nschonni** [comment]
Not sure if it is a a typo in your example or your code, but you have 2 dashes in front of your first item, so that wouldn't be a valid array of arrays in YAML
```yaml
outputFormatters:
    - markdownlint-cli2-formatter-default:
    - markdownlint-cli2-formatter-junit:
```
Also missing the trailing `:` to mark it as an empty array or to set the config as child keys https://github.com/DavidAnson/markdownlint-cli2/blob/main/doc/OutputFormatters.md

**Comment by DavidAnson** [comment]
Thanks, Nick, I noticed the double dash but not the missing colons!

**Comment by adam-moss** [comment]
Thanks for the replies!

I'm still failing to get this to work.  If I have

```yaml
---
config:
    default: true

customRules:
    - "markdownlint-rule-titlecase"

fix: true

noInlineConfig: true

outputFormatters:
    - markdownlint-cli2-formatter-default
    - markdownlint-cli2-formatter-junit
```

Then VS Code with the plugin complains with

```json
[{
	"resource": "/home/adam-moss/<redact>/.markdownlint-cli2.yaml",
	"owner": "_generated_diagnostic_collection_name_#4",
	"severity": 8,
	"message": "Incorrect type. Expected \"array\".",
	"source": "yaml-schema: markdownlint-cli2 configuration schema",
	"startLineNumber": 13,
	"startColumn": 7,
	"endLineNumber": 13,
	"endColumn": 42
}]
```

(repeated for the 2nd array entry)

If I use:

```yaml
outputFormatters:
   - - markdownlint-cli2-formatter-default
     - markdownlint-cli2-formatter-junit
````

Then VS Code is happy, but I get the issue described initially above.

If I use the suggested:

```yaml
outputFormatters:
    - markdownlint-cli2-formatter-default:
    - markdownlint-cli2-formatter-junit:
```

Then I get the aforementioned issue in VS Code, and 

```console
markdownlint-cli2 "**/*.md"
markdownlint-cli2 v0.5.1 (markdownlint v0.26.2)
Finding: **/*.md
Linting: 3 file(s)
Summary: 11 error(s)
TypeError: entry.slice is not a function or its return value is not iterable
    at /home/node_modules/markdownlint-cli2/markdownlint-cli2.js:108:46
    at Array.map (<anonymous>)
    at importOrRequireIdsAndParams (/home/node_modules/markdownlint-cli2/markdownlint-cli2.js:108:5)
    at async outputSummary (/home/node_modules/markdownlint-cli2/markdownlint-cli2.js:781:9)
    at async main (/home/node_modules/markdownlint-cli2/markdownlint-cli2.js:894:25)
    at async /home/node_modules/markdownlint-cli2/markdownlint-cli2.js:914:26
```

So I'm a bit unsure where I'm going wrong with this one.

**Comment by DavidAnson** [comment]
It looks like this is the appropriate YAML to create an array of arrays with a single string, each:

```
outputFormatters:
    - - markdownlint-cli2-formatter-default
    - - markdownlint-cli2-formatter-junit
```

Please try that with CLI2. Honestly, this shouldn't really work in the VS Code extension, because that runs on every update and shouldn't be doing things like writing to output files.

I'll have a look at blocking this in a future release.

**Comment by adam-moss** [comment]
That works, thank you!

I'm running this via the command line, the VS Code bit was the validation of the schema, although looking at it it doesn't appear to be writing any errors out to file in VS Code anyway, so perhaps there is nothing there for you to do ?

Interestingly enough the VS Code plugin does error out due to the `customRules` where-as the cli itself doesn't.

```console
[7:38:07 PM] INFO: Re-linting due to "/home/adam-moss/<redact>/.markdownlint-cli2.yaml" change.
[7:38:15 PM] ERROR: Exception while linting with markdownlint-cli2:
Error: Cannot find module 'markdownlint-rule-titlecase'
Require stack:
- /home/adam-moss/.vscode-insiders/extensions/davidanson.vscode-markdownlint-0.48.1/bundle.js
- /usr/share/code-insiders/resources/app/out/vs/loader.js
- /usr/share/code-insiders/resources/app/out/bootstrap-amd.js
- /usr/share/code-insiders/resources/app/out/bootstrap-fork.js
-  / Invalid host defined options
	at w (/home/adam-moss/.vscode-insiders/extensions/davidanson.vscode-markdownlint-0.48.1/bundle.js:37:6800)
	at async Promise.all (index 0)
	at async Promise.all (index 0)
	at async T (/home/adam-moss/.vscode-insiders/extensions/davidanson.vscode-markdownlint-0.48.1/bundle.js:37:9515)
	at async L (/home/adam-moss/.vscode-insiders/extensions/davidanson.vscode-markdownlint-0.48.1/bundle.js:37:13852)
[7:38:17 PM] ERROR: Exception while linting with markdownlint-cli2:
Error: Cannot find module 'markdownlint-rule-titlecase'
Require stack:
- /home/adam-moss/.vscode-insiders/extensions/davidanson.vscode-markdownlint-0.48.1/bundle.js
- /usr/share/code-insiders/resources/app/out/vs/loader.js
- /usr/share/code-insiders/resources/app/out/bootstrap-amd.js
- /usr/share/code-insiders/resources/app/out/bootstrap-fork.js
-  / Invalid host defined options
	at w (/home/adam-moss/.vscode-insiders/extensions/davidanson.vscode-markdownlint-0.48.1/bundle.js:37:6800)
	at async Promise.all (index 0)
	at async Promise.all (index 0)
	at async T (/home/adam-moss/.vscode-insiders/extensions/davidanson.vscode-markdownlint-0.48.1/bundle.js:37:9515)
	at async L (/home/adam-moss/.vscode-insiders/extensions/davidanson.vscode-markdownlint-0.48.1/bundle.js:37:13852)
[7:38:17 PM] ERROR: Exception while linting with markdownlint-cli2:
Error: Cannot find module 'markdownlint-rule-titlecase'
Require stack:
- /home/adam-moss/.vscode-insiders/extensions/davidanson.vscode-markdownlint-0.48.1/bundle.js
- /usr/share/code-insiders/resources/app/out/vs/loader.js
- /usr/share/code-insiders/resources/app/out/bootstrap-amd.js
- /usr/share/code-insiders/resources/app/out/bootstrap-fork.js
-  / Invalid host defined options
	at w (/home/adam-moss/.vscode-insiders/extensions/davidanson.vscode-markdownlint-0.48.1/bundle.js:37:6800)
	at async Promise.all (index 0)
	at async Promise.all (index 0)
	at async T (/home/adam-moss/.vscode-insiders/extensions/davidanson.vscode-markdownlint-0.48.1/bundle.js:37:9515)
	at async L (/home/adam-moss/.vscode-insiders/extensions/davidanson.vscode-markdownlint-0.48.1/bundle.js:37:13852)
```

That module is installed globally via `npm`, and the cli recognises it.

**Comment by DavidAnson** [comment]
Known issue: https://github.com/DavidAnson/vscode-markdownlint/issues/180

**Comment by adam-moss** [comment]
Yep, just spotted that one.  I think this issue can be closed.

Thank you very much for your kind assistance.
